client/src/main/java/io/pravega/client/auth/CredentialsCreator.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.auth;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.pravega.client.ClientConfig;
+import io.pravega.client.stream.impl.Credentials;
+import java.util.Map;
+import java.util.Properties;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+
+/**
+ * Helper class to extract credentials.
+ */
+public class CredentialsCreator {
```

Comment:
As per the main method in this class, this is not really a `CredentialsCreator`, this is a `CredentialsExtractor`, in which case, we should be able to get a new config by doing `CredentialsExtractor.extract()`.

Reply:
Fixed.